mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergFilterFactory.java

```
@@ -0,0 +1,159 @@
+      default:
+        throw new IllegalStateException("Unknown operator: " + leaf.getOperator());
+    }
+  }
+
+  private static Object leafToLiteral(PredicateLeaf leaf) {
+    switch (leaf.getType()) {
+      case LONG:
+      case BOOLEAN:
+      case STRING:
+      case FLOAT:
+        return leaf.getLiteral();
+      case DATE:
+        //Hive converts a Date type to a Timestamp internally when retrieving literal
+        return ((Timestamp) leaf.getLiteral()).toLocalDateTime().toLocalDate().toEpochDay();
+      case DECIMAL:
+        return BigDecimal.valueOf(((HiveDecimalWritable) leaf.getLiteral()).doubleValue());
+      case TIMESTAMP:
+        Timestamp timestamp = (Timestamp) leaf.getLiteral();
+        return timestamp.toInstant().getEpochSecond() * MICROS_PER_SECOND +
+                timestamp.getNanos() / NANOSECS_PER_MICROSEC;
+      default:
+        throw new IllegalStateException("Unknown type: " + leaf.getType());
+    }
+  }
+
+  private static List<Object> leafToLiteralList(PredicateLeaf leaf) {
+    switch (leaf.getType()) {
+      case LONG:
+      case BOOLEAN:
+      case FLOAT:
+      case STRING:
+        return leaf.getLiteralList();
+      case DATE:
+        List<Object> dateValues = leaf.getLiteralList();
+        dateValues.replaceAll(value -> ((Date) value).toLocalDate().toEpochDay());
+        return dateValues;
+      case DECIMAL:
+        List<Object> decimalValues = leaf.getLiteralList();
+        decimalValues.replaceAll(value -> BigDecimal.valueOf(((HiveDecimalWritable) value).doubleValue()));
+        return decimalValues;
+      case TIMESTAMP:
+        List<Object> timestampValues = leaf.getLiteralList();
+        timestampValues.replaceAll(value -> ((Timestamp) value).toInstant().getEpochSecond() * MICROS_PER_SECOND +
+                ((Timestamp) value).getNanos() / NANOSECS_PER_MICROSEC);
+        return timestampValues;
+      default:
+        throw new IllegalStateException("Unknown type: " + leaf.getType());
```

nit: better to throw unsupportedexception